### ISSUE TITLE
What to do with selection tuples?

### ISSUE THREAD

**Issue by cgreenberg** [salilab/pmi#145]
So in PMI1 a restraint is typically set up with:

``` python
x = Restraint(representation = r,
            selection_tuples=[('A',100,150),('B',200,235)],
             resolution=10)
```

Then the restraint internally does the selection. But now in PMI2 we have nice "handles" and also IMP Selection works great with all resolutions (including densities). We could now just have all restraints be set up like so:

``` python
x = Restraint(hiers = [molA[100:150],molB[200:235]],
            resolution=10)
```

So we'd get rid of selection tuples altogether. What do you think? @Pellarin ?

**Comment by Pellarin** [comment]
Yes, but we still need backcompatibility. How can we prevent to broke previous code?

**Comment by benmwebb** [comment]
Why not just pass a Selection object to each PMI restraint, rather than passing Selection-like parameters (e.g. resolution, residue indices) to each one?

**Comment by cgreenberg** [comment]
I'm all for that. Only problem is that many PMI restraints do the selecting for you - they read in data and select within the passed hierarchy. E.g. crosslink data or elastic networks on SSE elements. Maybe the way to do it is have PMI helper functions which create a selection from input data?

**Comment by benmwebb** [comment]
Perhaps... or we make Selection itself more flexible.

**Comment by cgreenberg** [comment]
Proposal after talking with @saltzberg : have all data parsers in IO (including `CrosslinkDatabase`, `DSSPParser`, ...) have a `select(hier,**kwargs)` function which: 1) selects particles in `hier` according to the underlying data (e.g. gives lists of lists of particle pairs for crosslinks) and 2) admits additional selection keywords. So you could, for example, create a customized XL selection with:

``` python
xldb = IMP.pmi.io.crosslink.CrossLinkDataBase()
xldb.create_set_from_file("xlinks.csv")
pps = xldb.select(hier,resolution=0,atom_type=IMP.atom.AtomType("NZ"))
r = IMP.pmi.restraints.CrossLinkingMassSpectrometryRestraint(particle_pair_lists=pps,
                                                             slope=0.0,
                                                             length=21.0,
                                                             label="XL")
```

Thoughts?

**Comment by benmwebb** [comment]
1) looks fine but 2) duplicates the existing Selection framework somewhat, so you'd end up writing a lot of duplicated code.

**Comment by cgreenberg** [comment]
I'd have it internally call IMP Selection and just pass the kwargs on.
